cpp/lib/include/DbConnection.h

```diff
@@ -153,10 +153,13 @@ class DbConnection {
     /** \note Converts the binary contents of "unescaped_string" into a form that can used as a string.
      *  \note This probably breaks for Sqlite if the string contains binary characters.
      */
-    std::string escapeString(const std::string &unescaped_string, const bool add_quotes = false);
+    std::string escapeString(const std::string &unescaped_string, const bool add_quotes = false, const bool return_null_on_empty_string=false);
     inline std::string escapeAndQuoteString(const std::string &unescaped_string) {
         return escapeString(unescaped_string, /* add_quotes = */true);
     }
+    inline std::string escapeAndQuoteStringOrNull(const std::string &unescaped_string) {
```

Comment:
I don't think this name is particularly good. I know it's a monstrosity, but how about escapeAndQuoteNonEmptyStringOrReturnNull?